core/src/sumeragi/main_loop.rs

```diff
@@ -383,40 +365,34 @@ impl Sumeragi {
         &self,
         state: &'state State,
         topology: &Topology,
-        genesis_account: &AccountId,
-        BlockCreated { block }: BlockCreated,
-        existing_voting_block: &mut Option<VotingBlock>,
-    ) -> Option<VotingBlock<'state>> {
-        if state.view().height() == 1 && block.header().height.get() == 1 {
-            // Consider our peer has genesis,
-            // and some other peer has genesis and broadcast it to our peer,
-            // then we can ignore such genesis block because we already has genesis.
-            // Note: `ValidBlock::validate` also checks it,
-            // but we don't want warning to be printed since this is correct behaviour.
-            return None;
-        }
+        BlockCreated {
+            transactions,
+            creation_time_ms,
+        }: BlockCreated,
+    ) -> VotingBlock<'state> {
+        let transactions = transactions
+            .into_iter()
+            .map(|stx| AcceptedTransaction(stx))
```

Comment:
```suggestion
            .map(|stx| AcceptedTransaction::accept(stx))
```

don't ever skip validation. The inner field of `AcceptedTransaction` should have been private